server/src/main/java/org/opensearch/index/query/IntervalsSourceProvider.java

```diff
@@ -465,7 +497,7 @@ public void writeTo(StreamOutput out) throws IOException {
         @Override
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
             builder.startObject(NAME);
-            builder.field("ordered", ordered);
+            builder.field("mode", mode);
```

Comment:
This is breaking, may be we could still keep `ordered` here, it corresponds well to the mode:
```
    ORDERED -> `ordered`: true
    UNORDERED(1) -> `ordered`: false
    UNORDERED_NO_OVERLAP(2) -> `ordered`: false
```

Reply:
@reta  how does this break since this is just generation.  The parser handles the bwc if a user was to send the `ordered` flag, should we be creating json with deprecated values?

Reply:
@mattweber thinking if there are any chances someone relies on `XContent` of this `IntervalsSourceProvider` (manipulate or consume JSON or something), it this case we could break them.

Reply:
@reta if that was the case they could use the correct version of es they are sending the json against.  Any new feature will break them (the regexp interval and wildcard max expansion PR's we have commited recently didn't do this)

Reply:
Fair enough, thanks @mattweber !